### PR TITLE
DG Migration Wrap V2 Adapter

### PR DESCRIPTION
**Pull request by FlattestWhite**
# Code Review Processes
## New Feature Review
Before submitting a pull request for new review, make sure the following is done:
* Design doc is created and posted here: [Insert Link]
* Code cleanliness and completeness is addressed via [guidelines](https://app.gitbook.com/@setprotocol-1/s/set/smart-contract-engineering/sc-code-review-process)

README Checks
- [] README has proper context for the reviewer to understand what the code includes, any important design considerations, and areas to pay more attention to

Code Checks
- [] Add explanatory comments. If there is complex code that requires specific context or understanding, note that in a comment
- [] Remove unncessary comments. Any comments that do not add additional context, information, etc. should be removed
- [] Add javadocs. 
- [] Scrub through the code for inconsistencies (e.g. removing extra spaces)
- [] Ensure there are not any .onlys in spec files

Broader Considerations
- [] Ensure variable, function and event naming is clear, consistent, and reflective for the scope of the code.
- [] Consider if certain pieces of logic should be placed in a different library, module

### CODE REVIEW

File: utils/deploys/deployExternal.ts [hunk 1]
```diff
@@ -233,6 +233,9 @@ import { PerpV2InsuranceFund__factory } from "../../typechain/factories/PerpV2In
 import { PerpV2AccountBalance__factory } from "../../typechain/factories/PerpV2AccountBalance__factory";
 import { PerpV2Exchange__factory } from "../../typechain/factories/PerpV2Exchange__factory";
 
+import { Dg } from "../contracts/dg";
+import { Dg__factory } from "../../typechain/factories/Dg__factory";
```
Comment (ncitron) [inline utils/deploys/deployExternal.ts:237]: Are you missing the bytecode for the DG token? Looks like typechain managed to create the factory contracts so maybe you forgot to commit something? Most of the time that dg.json file would include some bytecode along with the abi.

**Review by ncitron — COMMENTED** [review]
This is looking pretty good. One comment since I think your dg.json file is missing the bytecode. One thing we also usually do is an integration test to show that no only is the calldata generated correctly, but also that the contract interactions work as expected. Take a look at the agi migration to see how these type of tests usually work https://github.com/SetProtocol/set-protocol-v2/blob/master/test/integration/agiMigrationWrapModule.spec.ts

Also do you have the address of the new DG token?

**Comment by FlattestWhite** [comment]
> This is looking pretty good. One comment since I think your dg.json file is missing the bytecode. One thing we also usually do is an integration test to show that no only is the calldata generated correctly, but also that the contract interactions work as expected. Take a look at the agi migration to see how these type of tests usually work https://github.com/SetProtocol/set-protocol-v2/blob/master/test/integration/agiMigrationWrapModule.spec.ts
> 
> Also do you have the address of the new DG token?

Ahh right - yes I'll add a test testing the before and after state.

The new DG token address is here: https://etherscan.io/token/0x4b520c812e8430659fc9f12f6d0c39026c83588d

**Comment by FlattestWhite** [comment]
Screen Shot 2021-12-22 at 1 46 35 am
@ncitron need some advice - I've been trying to debug why the new integration test doens't work when it's being called through the wrap module. Do you have an idea why it's throwing this error?

**Comment by ncitron** [comment]
This error usually happens when a module calls `invoke` on a set token which causes the set token to make an arbitrary call (in this case it makes the wrapping call). This call is failing and causing the error. I would start by validating that you are interacting with the DG contract correctly.

**Comment by FlattestWhite** [comment]
pingaling @ncitron there's a deadline for this iirc

**Comment by cgewecke** [comment]
Closing in favor of #183 (please just ping if this should remain open)

**Pull request closed without merging**